test/e2e/reachability.go

```diff
@@ -381,3 +381,44 @@ func (r *Reachability) PrintSummary(printExpected bool, printObserved bool, prin
 		fmt.Printf("comparison:\n\n%s\n\n\n", comparison.PrettyPrint(""))
 	}
 }
+
+const (
+	NPEvalNONE    string = "<NONE>"
+	NPEvalAllow   string = "Allow"
+	NPEvalDrop    string = "Drop"
+	NPEvalIsolate string = "Isolate"
+	NPEvalReject  string = "Reject"
+)
+
+type NPEvaluationSpec struct {
+	Source      Pod
+	Destination Pod
+	NPName      string
+	Action      string
+}
+
+type NPEvaluation struct {
+	itemSet map[string]bool
+	Truth   []*NPEvaluationSpec
+}
+
+func NewNPEvaluation(pods []Pod) *NPEvaluation {
+	itemSet := map[string]bool{}
+	for _, pod := range pods {
+		itemSet[pod.String()] = true
+	}
+	return &NPEvaluation{
+		itemSet: itemSet,
+		Truth:   []*NPEvaluationSpec{},
+	}
+}
+
+func (e *NPEvaluation) Expect(from Pod, to Pod, npName string, action string) {
```

Comment:
I'd recommend returning `e` in this function, this way you can chain assertions in a less verbose way:
```go
NewNPEvaluation(...).Expect(...).Expect(...)
```